.github/workflows/pr-benchmarks.yml

```diff
@@ -0,0 +1,35 @@
+name: "PR Benchmark workflow"
```

Comment:
@algochoi How will other team members discover + extend the benchmarks?

I think it'd be helpful for maintainers to see https://algorand.github.io/go-algorand/dev/bench/ linked in the repo + understand  tooling limitations.

I'm imagining a README.md added to `.github/workflows/` (flexible on location) that outlines:
* Link to see historical benchmarks.
* Guidelines for adding benchmarks.  We've discussed a couple considerations like impact to build time and magnitude of performance degradation captured.

Reply:
I can put it in a README in the workflows folder (or alternatively in a Confluence/Google doc in a shared directory) - no objections there.

Reply:
@algochoi Sounds good - I'm flexible on location within the repo.  My preference is to version the doc in git (i.e. avoid Confluence and gdoc).

Reply:
@algochoi Resolving - Thanks for adding a README.